frontend/src/pages/Home.svelte

```diff
@@ -32,13 +37,113 @@ let paths: {
   JSON.parse(window.localStorage.getItem("BOT_SEARCH_PATHS") || "[]"),
 );
 
+let botpackNotifIds: { [repo: string]: string } = {};
+
+function updateBotpack(repoName: string) {
+  const notifId = botpackNotifIds[repoName];
+  if (notifId) {
+    toast.dismiss(notifId);
+  }
+
+  const details = paths.find((x) => x.repo === repoName);
+  if (details?.repo && details.tagName) {
+    const tId = toast.loading(`Updating ${repoName}...`, {
+      position: "bottom-right",
+    });
+
+    App.UpdateBotpack(details.repo, details.installPath, details.tagName)
+      .then((newTagName) => {
+        details.tagName = newTagName;
+        toast.success(`${repoName} updated successfully`, {
+          id: tId,
+          position: "bottom-right",
+          duration: 3000,
+        });
+      })
+      .catch((error) => {
+        console.error(error);
+        toast.error(`Failed to update ${repoName}: ${error}`, {
+          id: tId,
+          position: "bottom-right",
+          duration: 10000,
+        });
+      });
+  }
+}
+
+async function CheckForBotpackUpdates() {
```

Comment:
this function is async but we never use await?

Reply:
JavaScript async is different from Rust async. An async function starts as soon as you call it, the await just blocks the current thread. If you don't want the function to block the main thread, then you just don't await it. The function returns nothing we need to handle, so it works.